vmware-tanzu/tanzu-framework
# Support management package upgrades as part of `management-cluster` upgrade

### What this PR does / why we need it

- Support management package upgrades as part of `management-cluster` upgrade
- Install version specific packages instead of latest available packages
- Add label to `PackageInstall` for all management packages

**Approach:**
- Management Cluster Creation:
    - Install management package repository (v0.21.0)
    - Install TKG composite package (v0.21.0) [use specific version constraints with `0.21.0`]
    - Wait for all management packages to get reconciled successfully

- Management Cluster Upgrade:
    - Update management package repository to new repository version (v0.22.0)
    - Upgrade TKG composite package (v0.22.0) [update to specific version constraints with `0.22.0`]
    - Wait for all management packages to get reconciled successfully after upgrade 


### Which issue(s) this PR fixes


Fixes #2137

### Describe testing done for PR

1. Build management package repository (`v0.21.0`)
2. Create management cluster with `package-based-lcm` feature-flag enabled
3. Verify that management cluster gets created successfully with all management packages reconciled successfully
4. Rebuild the management package repository with new package versions (`v0.22.0`)
5. Run the management cluster upgrade command
6. Verify that management cluster packages gets upgraded to new versions successfully



### Release note

```release-note
package-based-lcm: Support management package upgrades as part of `management-cluster` upgrade
```

### PR Checklist



      


### Additional information

#### Special notes for your reviewer
